ports/nrf/boards/SUPERMINI_NRF52840 board definition + machine_bitstream.

jkorte-dev opened the pull request:
The Supermini NRF52840 is a clone of the NiceNano board based on the Pro Micro layout. The board definition includes a machine_bitstream driver to drive NeoPixel leds.


### Summary

the Supermini NRF52840 is a quite popular and affordable board available from various resellers in Asia / Europe / US.
The board definition has been extended to work for Adafruit Feather nRF52840 Express
as well. I added [machine.bitstream](https://docs.micropython.org/en/latest/library/machine.html#machine.bitstream) for nrf52 to be able to drive Neopixel LEDs . The machine_bitstream.c code is based on machine_bitstream of stm32 and w600 port and was  adopted for the nrf52840. The driver should work with any nrf52 board (confirmed to work on a Adafruit Feather nRF52840 Express). The bitstream driver is includes here but  can extract it to be used with other nrf52 series boards.


### Testing

I tested basic functionality of the board. The neopixel driver was tested with a bunch of Neopixel LEDs/ LED matrix and the timing verified with a simple logic analyzer.

### Trade-offs and Alternatives

Please let me know if I have overseen any trade-offs.

jkorte-dev committed:
ports/nrf/boards: SUPERMINI_NRF52840 board definition.

The Supermini NRF52840 is a clone of the NiceNano board.
The board definition includes a machine_bitstream driver
to drive NeoPixel leds.

Signed-off-by: Joerg Korte <joerg.korte@gmail.com>

JDRBoston commented:
I'll be happy to test your build in controlling the NeoPixel on the Adafruit Feather nRF52840 board. Do you know how your pin mappings might apply to that board? There, the NeoPixel resides on GPIO P0.16? pin = Pin(16. Pin.OUT) indicates that it isn't a valid pin identifier.

jkorte-dev committed:
ports/nrf/boards: SUPERMINI_NRF52840 board definition.

Update pins.csv to support pins of Adafruit Feather nRF52840 Express.
Update mpconfigboard.h for latest micropython.

Signed-off-by: Joerg Korte <joerg.korte@gmail.com>

jkorte-dev committed:
ports/nrf/boards/SUPERMINI_NRF52840/pins.csv: fixed P1_04.

Update pins.csv to fix definition for P1_04.

Signed-off-by: Joerg Korte <joerg.korte@gmail.com>
